diff -up gnome-mplayer-1.0.0/generic/src/Makefile.am.av gnome-mplayer-1.0.0/generic/src/Makefile.am
--- gnome-mplayer-1.0.0/generic/src/Makefile.am.av	2011-02-16 23:54:09.000000000 +0100
+++ gnome-mplayer-1.0.0/generic/src/Makefile.am	2011-02-20 17:19:03.000000000 +0100
@@ -97,7 +97,7 @@ libgnome_mplayer_properties_page_so_CFLA
     $(GIO_CFLAGS) \
     $(AM_CFLAGS)
     
-libgnome_mplayer_properties_page_so_LDFLAGS = -avoid-version -shared
+libgnome_mplayer_properties_page_so_LDFLAGS = -shared
 
 libgnome_mplayer_properties_page_so_LDADD = \
 	$(NAUTILUS_LIBS) \
diff -up gnome-mplayer-1.0.0/generic/src/Makefile.in.av gnome-mplayer-1.0.0/generic/src/Makefile.in
--- gnome-mplayer-1.0.0/generic/src/Makefile.in.av	2011-02-17 23:42:56.000000000 +0100
+++ gnome-mplayer-1.0.0/generic/src/Makefile.in	2011-02-20 17:18:41.000000000 +0100
@@ -387,7 +387,7 @@ gnome_mplayer_LDADD = \
 @HAVE_NAUTILUS_TRUE@    $(GIO_CFLAGS) \
 @HAVE_NAUTILUS_TRUE@    $(AM_CFLAGS)
 
-@HAVE_NAUTILUS_TRUE@libgnome_mplayer_properties_page_so_LDFLAGS = -avoid-version -shared
+@HAVE_NAUTILUS_TRUE@libgnome_mplayer_properties_page_so_LDFLAGS = -shared
 @HAVE_NAUTILUS_TRUE@libgnome_mplayer_properties_page_so_LDADD = \
 @HAVE_NAUTILUS_TRUE@	$(NAUTILUS_LIBS) \
 @HAVE_NAUTILUS_TRUE@	$(GTK_LIBS) \
