Index: src/playlist.c
===================================================================
--- src/playlist.c	(wersja 1092)
+++ src/playlist.c	(wersja 1093)
@@ -1008,8 +1008,8 @@
 
         gtk_paned_pack2(GTK_PANED(pane), plvbox, TRUE, TRUE);
         adjust_paned_rules();
-
-        // Style doesn't seem to have this property
+		
+		move_pane_position = TRUE;
         gtk_widget_style_get(pane, "handle-size", &handle_size, NULL);
         if (vertical_layout) {
             gtk_widget_set_size_request(plvbox, -1, 150);
Index: src/thread.c
===================================================================
--- src/thread.c	(wersja 1092)
+++ src/thread.c	(wersja 1093)
@@ -722,7 +722,7 @@
     g_idle_add(set_progress_value, idledata);
     g_idle_add(set_progress_text, idledata);
     g_idle_add(set_media_info, idledata);
-    //g_idle_add(set_window_visible, idledata);
+    g_idle_add(set_window_visible, idledata);
 
     if (mplayer_bin == NULL || !g_file_test(mplayer_bin, G_FILE_TEST_EXISTS)) {
         argv[arg++] = g_strdup_printf("mplayer");
Index: src/main.c
===================================================================
--- src/main.c	(wersja 1092)
+++ src/main.c	(wersja 1093)
@@ -156,7 +156,7 @@
     GtkRecentData *recent_data;
     GtkRecentInfo *recent_info;
 #endif
-
+	
     if (gtk_list_store_iter_is_valid(playliststore, playiter)) {
         gtk_tree_model_get(GTK_TREE_MODEL(playliststore), playiter, ITEM_COLUMN, &uri,
                            DESCRIPTION_COLUMN, &title,
@@ -541,6 +541,7 @@
     disable_cover_art_fetch = FALSE;
     mixer = NULL;
     fullscreen = 0;
+	move_pane_position = FALSE;
 
     sa.sa_handler = hup_handler;
     sigemptyset(&sa.sa_mask);
Index: src/gui.c
===================================================================
--- src/gui.c	(wersja 1092)
+++ src/gui.c	(wersja 1093)
@@ -60,8 +60,8 @@
 
 void adjust_paned_rules()
 {
+	return;
     if (!idledata->videopresent) {
-
         g_object_ref(vbox);
         gtk_container_remove(GTK_CONTAINER(pane), GTK_PANED(pane)->child1);
         gtk_paned_pack1(GTK_PANED(pane), vbox, FALSE, FALSE);
@@ -81,7 +81,7 @@
 {
     gint position;
 
-    g_object_get(pane, "position", &position, NULL);
+	g_object_get(pane, "position", &position, NULL);
     g_object_ref(vbox);
     gtk_container_remove(GTK_CONTAINER(pane), vbox);
     gtk_paned_pack1(GTK_PANED(pane), vbox, TRUE, TRUE);
@@ -466,10 +466,14 @@
 
 gboolean set_window_visible(void *data)
 {
-
+	//gtk_check_menu_item_set_active(GTK_CHECK_MENU_ITEM(menuitem_view_playlist),FALSE);
     if (GTK_IS_WIDGET(fixed)) {
-        gtk_widget_show_all(fixed);
-    }
+		if (vertical_layout) {
+			gtk_widget_show(fixed);
+		} else {
+			gtk_widget_show(vbox);
+		}    
+	}
     return FALSE;
 }
 
@@ -794,6 +798,7 @@
     IdleData *idle = (IdleData *) data;
     gint total_height = 0;
     gint total_width = 0;
+	gint handle_size;
     GTimeVal currenttime;
 
     if (GTK_IS_WIDGET(window)) {
@@ -838,8 +843,15 @@
 
                         total_width = idle->width;
                         if (GTK_IS_WIDGET(plvbox) && GTK_WIDGET_VISIBLE(plvbox)) {
+							gtk_widget_style_get(pane, "handle-size", &handle_size, NULL);
                             total_height += plvbox->allocation.height;
-                            total_width = idle->width + plvbox->allocation.width;
+                            total_width = idle->width + plvbox->allocation.width + handle_size;
+							if (vertical_layout) {
+								gtk_paned_set_position(GTK_PANED(pane),idle->height);
+							} else {
+								gtk_paned_set_position(GTK_PANED(pane),idle->width);
+							}
+							move_pane_position = TRUE;
                         }
 
                         gtk_window_resize(GTK_WINDOW(window), total_width, total_height);
@@ -1272,6 +1284,7 @@
 
     gdouble movie_ratio, window_ratio;
     gint new_width, new_height;
+	gint handle_size;
 
 
     if (actual_x > 0 && actual_y > 0) {
@@ -1292,14 +1305,29 @@
         } else {
             if (movie_ratio > window_ratio) {
                 //printf("movie %lf > window %lf\n",movie_ratio,window_ratio);
-                new_width = allocation->width;
-                new_height = allocation->width / movie_ratio;
+				new_width = allocation->width;
+				new_height = allocation->width / movie_ratio;
             } else {
                 //printf("movie %lf < window %lf\n",movie_ratio,window_ratio);
-                new_height = allocation->height;
-                new_width = allocation->height * movie_ratio;
+				new_height = allocation->height;
+				new_width = allocation->height * movie_ratio;
             }
         }
+		
+		//printf("new_width %i new_height %i\n",new_width, new_height);
+		if (move_pane_position) {
+			if (gtk_check_menu_item_get_active(GTK_CHECK_MENU_ITEM(menuitem_view_playlist))) {
+				gtk_widget_style_get(pane, "handle-size", &handle_size, NULL);
+				if (vertical_layout) {
+						gtk_paned_set_position(GTK_PANED(pane),idledata->height);
+				} else {
+						gtk_paned_set_position(GTK_PANED(pane),idledata->width);
+				}
+			}
+			move_pane_position = FALSE;
+		}
+		printf("new_width %i new_height %i\n",new_width, new_height);
+		
         gtk_widget_set_size_request(drawing_area, new_width, new_height);
         gtk_widget_set_size_request(fixed, allocation->width, allocation->height);
         idledata->x = (allocation->width - new_width) / 2;
Index: src/common.h
===================================================================
--- src/common.h	(wersja 1092)
+++ src/common.h	(wersja 1093)
@@ -301,6 +301,7 @@
 gboolean load_tracks_from_gpod;
 gchar *gpod_mount_point;
 gboolean disable_cover_art_fetch;
+gboolean move_pane_position;
 
 gboolean remember_loc;
 gint loc_window_x;
