--- gnome-mplayer.schemas.flatvolume	2009-06-05 18:23:16.000000000 +0200
+++ gnome-mplayer.schemas	2009-06-05 20:30:08.000000000 +0200
@@ -474,7 +474,7 @@
       <applyto>/apps/gnome-mplayer/preferences/use_pulse_flat_volume</applyto>
       <owner>gnome-mplayer</owner>
       <type>bool</type>
-      <default>FALSE</default>
+      <default>TRUE</default>
       <locale name="C">
         <short>Set to TRUE if using with pulseaudio 0.9.15</short>
         <long>PulseAudio started using a flat volume option by default in 0.9.15 (you can disable however)</long>
