Index: src/dbus-interface.h
===================================================================
--- src/dbus-interface.h	(wersja 1365)
+++ src/dbus-interface.h	(wersja 1366)
@@ -30,5 +30,6 @@
 #include "support.h"
 
 DBusConnection *connection;
-uint cookie;
+uint ss_cookie;
+uint sm_cookie;
 gboolean dbus_hookup(gint windowid, gint controlid);
Index: src/dbus-interface.c
===================================================================
--- src/dbus-interface.c	(wersja 1365)
+++ src/dbus-interface.c	(wersja 1366)
@@ -1163,17 +1163,9 @@
 
 void dbus_unhook()
 {
-
-    DBusMessage *message;
-
     if (connection != NULL) {
-        message =
-            dbus_message_new_method_call("org.gnome.ScreenSaver", "/org/gnome/ScreenSaver",
-                                         "org.gnome.ScreenSaver", "UnInhibit");
-        dbus_message_append_args(message, DBUS_TYPE_INT32, &cookie, DBUS_TYPE_INVALID);
-        dbus_connection_send(connection, message, NULL);
-        dbus_message_unref(message);
-        // dbus_connection_close(connection);
+		dbus_enable_screensaver();
+		// dbus_connection_close(connection);
         dbus_connection_unref(connection);
         connection = NULL;
     }
@@ -1184,13 +1176,22 @@
     DBusMessage *message;
 
     if (connection != NULL) {
-        message =
-            dbus_message_new_method_call("org.gnome.ScreenSaver", "/org/gnome/ScreenSaver",
-                                         "org.gnome.ScreenSaver", "UnInhibit");
-        dbus_message_append_args(message, DBUS_TYPE_INT32, &cookie, DBUS_TYPE_INVALID);
+	    message =
+	        dbus_message_new_method_call("org.gnome.ScreenSaver", "/org/gnome/ScreenSaver",
+	                                     "org.gnome.ScreenSaver", "UnInhibit");
+        dbus_message_append_args(message, DBUS_TYPE_INT32, &ss_cookie, DBUS_TYPE_INVALID);
         dbus_connection_send(connection, message, NULL);
         dbus_message_unref(message);
-    }
+
+	    message =
+	        dbus_message_new_method_call("org.gnome.SessionManager", "/org/gnome/SessionManager",
+	                                     "org.gnome.SessionManager", "UnInhibit");
+        dbus_message_append_args(message, DBUS_TYPE_INT32, &sm_cookie, DBUS_TYPE_INVALID);
+        dbus_connection_send(connection, message, NULL);
+        dbus_message_unref(message);
+
+
+	}
 }
 
 void dbus_disable_screensaver()
@@ -1200,6 +1201,7 @@
     DBusMessage *message;
     const gchar *app;
     const gchar *reason;
+	gint flags;
 
     if (connection != NULL) {
         dbus_error_init(&error);
@@ -1211,12 +1213,34 @@
         dbus_message_append_args(message, DBUS_TYPE_STRING, &app, DBUS_TYPE_STRING, &reason,
                                  DBUS_TYPE_INVALID);
         reply_message = dbus_connection_send_with_reply_and_block(connection, message, 200, &error);
+	
         if (reply_message) {
-            dbus_message_get_args(reply_message, &error, DBUS_TYPE_INT32, &cookie, NULL);
+            dbus_message_get_args(reply_message, &error, DBUS_TYPE_INT32, &ss_cookie, NULL);
             dbus_message_unref(reply_message);
         }
-
+		
         dbus_message_unref(message);
         dbus_error_free(&error);
+
+			
+	    message =
+	        dbus_message_new_method_call("org.gnome.SessionManager", "/org/gnome/SessionManager",
+	                                     "org.gnome.SessionManager", "Inhibit");
+	    app = g_strdup_printf("gnome-mplayer");
+	    reason = g_strdup_printf("playback");
+		flags = 8;
+		dbus_message_append_args(message, DBUS_TYPE_STRING, &app, DBUS_TYPE_UINT32, &(idledata->windowid), 
+		                         DBUS_TYPE_STRING, &reason, DBUS_TYPE_UINT32, &flags,
+	                             DBUS_TYPE_INVALID);
+	    reply_message = dbus_connection_send_with_reply_and_block(connection, message, 200, &error);
+	
+	    if (reply_message) {
+	        dbus_message_get_args(reply_message, &error, DBUS_TYPE_INT32, &sm_cookie, NULL);
+	        dbus_message_unref(reply_message);
+	    }
+	
+	    dbus_message_unref(message);
+	    dbus_error_free(&error);
+
     }
 }
