Index: src/thread.c
===================================================================
--- src/thread.c	(wersja 1815)
+++ src/thread.c	(wersja 1816)
@@ -187,6 +187,7 @@
     if (strstr(mplayer_output->str, "Failed to open") != NULL) {
         if (strstr(mplayer_output->str, "LIRC") == NULL &&
             strstr(mplayer_output->str, "/dev/rtc") == NULL &&
+            strstr(mplayer_output->str, "VDPAU") == NULL &&
             strstr(mplayer_output->str, "registry file") == NULL) {
             // error_msg = g_strdup(mplayer_output->str);
             if (strstr(mplayer_output->str, "<") == NULL && strstr(mplayer_output->str, ">") == NULL
