Index: src/libgmlib/gm_parse.h
===================================================================
--- src/libgmlib/gm_parse.h	(revision 1545)
+++ src/libgmlib/gm_parse.h	(working copy)
@@ -29,6 +29,7 @@
 #else
 #include <stdio.h>
 #endif
+#include <string.h>

 #ifdef __cplusplus
 extern "C" {
Index: src/libgmlib/gm_parse.c
===================================================================
--- src/libgmlib/gm_parse.c	(revision 1545)
+++ src/libgmlib/gm_parse.c	(working copy)
@@ -159,6 +159,7 @@
 			
 #else
     FILE *fp;
+	gchar **parse;
     gchar *file = NULL;
 
     file = g_strndup(uri, 4);
